src/fsharp/InternalCollections.fsi

```diff
@@ -44,21 +44,30 @@ namespace Internal.Utilities.Collections
     new : keepStrongly:int 
             * areSame:('Key * 'Key -> bool) 
             * ?isStillValid:('Key * 'Value -> bool)
-            * ?areSameForSubsumption:('Key * 'Key -> bool) 
+            * ?areSimilar:('Key * 'Key -> bool) 
```

Comment:
A XML Doc comment would be great here explaining what "similarity" means here and what it will be used for.